Stable with tui improvements

joshuef opened the pull request:
## Description

reviewpad:summary

dirvine commented on sn_node/src/node/messaging/anti_entropy.rs:
```diff
@@ -296,41 +296,13 @@ impl MyNode {
-                .ae_backoff_cache
-                .insert((*peer, ExponentialBackoff::default()));
-        }
+    async fn wait_for_backoff(&mut self) {
```
Would it not be possible to take a duration in and then return the duration in case the message is still not sent. i.e. 

```async fn wait_for_backoff(&mut self, Duration d) -> Duration ```
So we try to send and if not we run again with Duration * 2 up to a value where we stop?

maqi replied:
This is in the AE flow, not within the send flow. 
So basically each `send` itself are correctly completed. Just the msg incoming holding incorrect SAP.

To do what you described, we also need to modify the cmd send cnum to pass the Duration around.
Also, the current send_failure doesn't give out the `Duration` ?

dirvine replied:
Ah right, Sorry @maqi for the noise. 
If Ae then I really don't think we should have a cache to wait for anything as the point is to return errors if the SAP is wrong. The sender can resend with correct SAP to us. Maybe we need a huddle on this one?

maqi replied:
yeah sure, left a message in slack

joshuef commented:
@maqi it's not clear to me _why_ we're removing this? What does hold off randomly get us vs this cache?

RolandSherwin committed:
chore: upgrade service manager crate

RolandSherwin committed:
feat(launchpad): add discord username scene

RolandSherwin committed:
feat(launchpad): have customizable footer

RolandSherwin committed:
feat(launchpad): use escape to exit input screen and restore old value

RolandSherwin committed:
refactor(launchpad): use safe data dir to store configs

RolandSherwin committed:
feat(launchpad): store discord username to disk

RolandSherwin committed:
fix: use fixed size popups

joshuef committed:
fix(launchpad): prevent loops from terminal/sudo relaunching

RolandSherwin committed:
chore: use consistent border styles

jacderida committed:
docs: update cli and readme for user-mode services

The node manager documentation is updated in the README and the CLI for the introduction of
user-mode services.

The README also needed a little update for the `local` subcommand.

jacderida committed:
feat: run safenode services in user mode

The `Systemd` and `Launchd` service managers support running services in user mode, and the
`service-manager-rs` crate already accommodates it, so we can support this scenario.

Services are managed in user mode depending on whether the node manager is running elevated or not.
In some cases, elevated privileges are still required, because not all service managers support
user-mode services. For example, Alpine uses OpenRC, which does not have support. A service running
in user-mode requires an active user session. This may not always be appropriate, particularly for
advanced users who want to leave nodes running permanently in the background.

The RPC daemon and the faucet services are still defined to run as system-wide services and they
don't support running in user-mode, because it doesn't really make sense to require an active user
session for these services.

joshuef committed:
fix(launchpad): prevent mac opening with sudo

Add context to some error messages

RolandSherwin committed:
feat(launchpad): allow users to input disk space to allocate

RolandSherwin committed:
feat(launchpad): initial automatic resource allocation logic

RolandSherwin committed:
chore(launchpad): add more alternative keybinds

RolandSherwin committed:
chore(launchpad): highlight the table in green if we're currently running

RolandSherwin committed:
chore: change trigger resource allocation input box keybind

RolandSherwin committed:
chore: move helper text inside popup

jacderida committed:
chore: change terminal launch behaviour

On Linux, the `TERM` variable is not really appropriate for selecting which terminal is in use, so
this has been removed. It can be set to a completely different value, like `xterm-256color`, which
does not necessarily correspond to the running terminal. The `sudo` was also removed from the
terminal launch mechanism.

On the Windows front, I updated Windows Terminal to not use the `/c` argument. It simply just passes
the command to the terminal.

Also fixed another couple of issues:
* Compile the node manager correctly on Windows.
* The use of `<service>` in doc comments was being flagged as an unclosed HTML tag.

RolandSherwin committed:
fix(launchpad): check if component is active before handling events

mazzi committed:
refactor(launchpad): removing redudnat for loops
